src/Acuminator/Acuminator.Analyzers/StaticAnalysis/EventHandlerModifier/EventHandlerProtectedModifierFix.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System.Composition;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CodeActions;
+using System.Collections.Generic;
+using System.Threading;
+using System.Linq;
+using Microsoft.CodeAnalysis.Text;
+using Acuminator.Analyzers.StaticAnalysis.PrivateEventHandlers;
+
+namespace Acuminator.Analyzers.StaticAnalysis.EventHandlerModifier
+{
+	[ExportCodeFixProvider(LanguageNames.CSharp), Shared]
+	public class EventHandlerProtectedModifierFix : PXCodeFixProvider
+	{
+		public override ImmutableArray<string> FixableDiagnosticIds { get; } =
+			ImmutableArray.Create(
+				Descriptors.PX1077_EventHandlersShouldNotBePrivate.Id,
+				Descriptors.PX1077_EventHandlersShouldBeProtectedVirtual.Id,
+				Descriptors.PX1077_EventHandlersShouldNotBeExplicitInterfaceImplementations.Id);
+
+		protected override Task RegisterCodeFixesForDiagnosticAsync(CodeFixContext context, Diagnostic diagnostic)
+		{
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+			var registerCodeFix = DiagnosticUtils.IsFlagSet(diagnostic, DiagnosticProperty.RegisterCodeFix);
+			var isContainingTypeSealed = DiagnosticUtils.IsFlagSet(diagnostic, PrivateEventHandlers.DiagnosticProperty.IsContainingTypeSealed);
+
+			if (!registerCodeFix)
+			{
+				return Task.CompletedTask;
+			}
+
+			var accessibilityModifier = isContainingTypeSealed
+				? SyntaxKind.PublicKeyword
+				: SyntaxKind.ProtectedKeyword;
+
+			var addVirtualModifier = !isContainingTypeSealed;
+
+			var modifierFormatArg = EventHandlerModifierAnalyzer.GetModifierFormatArg(accessibilityModifier, addVirtualModifier);
+
+			var makeProtectedTitle = nameof(Resources.PX1077Fix).GetLocalized(modifierFormatArg).ToString();
+
+
+			context.CancellationToken.ThrowIfCancellationRequested();
+
+
+			var codeFixAction = CodeAction.Create(
+				makeProtectedTitle,
+				cToken => ChangeAccessibilityModifierAsync(context.Document, context.Span, accessibilityModifier, addVirtualModifier, cToken),
+				equivalenceKey: makeProtectedTitle);
```

Comment:
`makeProtectedTitle` shouldn't be used as `equivalenceKey` for code action in this case because it contains format args.
Having a string which is different depending on the format arguments as `equivalenceKey` will prevent the batch fixer from applying this code fix to other places with the same error when the user clicks "Fix All" in VS.

This will happen because equivalence keys of the reported diagnostics will be different and Roslyn will assume that this code fix can't be applied to another diagnostic. 
 
To fix this pass the format template string without arguments as `equivalenceKey`:
```suggestion
				equivalenceKey: Resources.PX1077Fix);
```

Reply:
Done. Thanks.